src/assets/resolver/Resolver.ts

```diff
@@ -506,6 +506,25 @@ export class Resolver
             // loop through all the srcs and generate a resolve asset for each src
             const resolvedAssets: ResolvedAsset[] = [];
 
+            // Helper function to parse a URL string using registered parsers
+            const parseUrl = (url: string): ResolvedAsset =>
+            {
+                let result = { src: url } as ResolvedAsset;
+
+                for (let i = 0; i < this._parsers.length; i++)
+                {
+                    const parser = this._parsers[i];
+
+                    if (parser.test(url))
+                    {
+                        result = parser.parse(url);
+                        break;
+                    }
+                }
+
+                return result;
```

Comment:
Can we reduce this?
```suggestion
                const parser = this._parsers.find((p) => p.test(url));
                return parser?.parse(url);
```

Reply:
Yeah, no worries. Lots of older code in the project. This is not performance critical code, so we don't need to be quite as verbose.